src/librustc/ty/context.rs

```diff
@@ -384,6 +390,30 @@ impl<'tcx> TypeckTables<'tcx> {
     pub fn upvar_capture(&self, upvar_id: ty::UpvarId) -> ty::UpvarCapture<'tcx> {
         self.upvar_capture_map[&upvar_id]
     }
+
+    /// Validate that a NodeId can safely be converted to an ItemLocalId for
```

Comment:
When is this function called? It seems to be called on some random HIR id accesses.